central/processlisteningonport/store/postgres/full_store.go

```diff
@@ -38,6 +44,8 @@ const getByDeploymentStmt = "SELECT plop.serialized, " +
 	"ON plop.processindicatorid = proc.id " +
 	"WHERE plop.deploymentid = $1 AND plop.closed = false"
 
+// const getClusterAndNamespaceStmt = "select namespace, clusterid from deployments where id = $1"
```

Comment:
The idea here and with `checkPermission` and `checkPermissionsForRows` is to look up the cluster and namespace from the deploymentId and check if the user has access to the cluster and namespace. However, I commented this out as I wasn't sure what to do if nothing was returned. It could be a problem if nothing is returned and that is not considered an error and later in the execution data about listening endpoints to which the user does not have access to is returned. This is unlikely as if the deployment doesn't exist in the deployments table, there shouldn't be any listening endpoints for that deployment. Returning an error if no deployments are fetched also didn't seem correct.

Reply:
I am now using this code. If nothing is returned, GetProcessesListeningOnPorts returns with nil, nil.